server/src/main/java/org/apache/druid/server/coordinator/HttpLoadQueuePeon.java

```diff
@@ -65,7 +66,7 @@
 
 /**
  */
-public class HttpLoadQueuePeon extends LoadQueuePeon
+public class HttpLoadQueuePeon implements LoadQueuePeon
```

Comment:
Change summary:
Used `QueuedSegment` instead of `SegmentHolder` and implemented new `LoadQueuePeon` methods for cancellation and loading with type.

Reply:
Reverted.